File: java/app/src/main/java/githubtododemo/githubclient/repos/ReposRequestBuilder.java
package githubtododemo.githubclient.repos;

import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import githubtododemo.githubclient.repos.item.WithOwnerItemRequestBuilder;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /repos
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class ReposRequestBuilder extends BaseRequestBuilder {
    /**
     * Gets an item from the githubtododemo.githubclient.repos.item collection
     * @param owner The account owner of the repository. The name is not case sensitive.
     * @return a {@link WithOwnerItemRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public WithOwnerItemRequestBuilder byOwner(@jakarta.annotation.Nonnull final String owner) {
        Objects.requireNonNull(owner);
        final HashMap<String, Object> urlTplParams = new HashMap<String, Object>(this.pathParameters);
        urlTplParams.put("owner", owner);
        return new WithOwnerItemRequestBuilder(urlTplParams, requestAdapter);
    }
    /**
     * Instantiates a new {@link ReposRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ReposRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/repos", pathParameters);
    }
    /**
     * Instantiates a new {@link ReposRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ReposRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/repos", rawUrl);
    }
}

File: php/GitHub/Repos/ReposRequestBuilder.php
<?php

namespace Baywet\Githubtododemo\Github\Repos;

use Baywet\Githubtododemo\Github\Repos\Item\WithOwnerItemRequestBuilder;
use Microsoft\Kiota\Abstractions\BaseRequestBuilder;
use Microsoft\Kiota\Abstractions\RequestAdapter;

/**
 * Builds and executes requests for operations under /repos
*/
class ReposRequestBuilder extends BaseRequestBuilder 
{
    /**
     * Gets an item from the baywet.githubtododemo.github.repos.item collection
     * @param string $owner Unique identifier of the item
     * @return WithOwnerItemRequestBuilder
    */
    public function byOwner(string $owner): WithOwnerItemRequestBuilder {
        $urlTplParams = $this->pathParameters;
        $urlTplParams['owner'] = $owner;
        return new WithOwnerItemRequestBuilder($urlTplParams, $this->requestAdapter);
    }

    /**
     * Instantiates a new ReposRequestBuilder and sets the default values.
     * @param array<string, mixed>|string $pathParametersOrRawUrl Path parameters for the request or a String representing the raw URL.
     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
    */
    public function __construct($pathParametersOrRawUrl, RequestAdapter $requestAdapter) {
        parent::__construct($requestAdapter, [], '{+baseurl}/repos');
        if (is_array($pathParametersOrRawUrl)) {
            $this->pathParameters = $pathParametersOrRawUrl;
        } else {
            $this->pathParameters = ['request-raw-url' => $pathParametersOrRawUrl];
        }
    }

}
